run.py

```diff
@@ -11,6 +11,28 @@
 from mig_dx_api._dataset import DatasetOperations
 from google.cloud import bigquery
 
+def get_target_installation(installations: list[Installation], target_installation_id: str) -> Installation:
+    if len(installations) == 0:
+        raise Exception("No valid installations found")
+    elif len(installations) == 1:
+        target_install = installations[0]
+        # error if the target installation id doesn't match the only existing installation
+        if target_installation_id:
+            if target_install.movement_app_installation_id != int(target_installation_id):
+                raise Exception(f"Installation {target_installation_id} not found")
+        return target_install
+    elif len(installations) > 1 and target_installation_id is None:
```

Comment:
nit: `len(installations) > 1` is unnecessary here as we have already checked lengths 0 and 1. Personally, in order to make this parallel the `elif len(installations) == 1` block, I'd put the `else` here and then check for `target_installation_id` within the else and raise the Exception there if necessary.